### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD029`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#743]
### Rule details

Add a `consistent-ordered-list-style` rule based on markdownlint's MD029.

Support the `one`, `one_or_ordered`, `ordered`, and `zero` styles, with `one_or_ordered` as the default. The rule should report ordered list item prefixes that do not match the configured style.

Autofix is out of scope for the initial implementation.

### What type of rule is this?

Enforces a consistent style or user preference

### Example code

```markdown
<!-- Incorrect -->

1. First item
3. Second item

<!-- Correct: one -->

1. First item
1. Second item

<!-- Correct: ordered -->

1. First item
2. Second item
```

### Prior Art

- markdownlint: [`MD029` - Ordered list item prefix](https://github.com/DavidAnson/markdownlint/blob/main/doc/md029.md)
